iroha/src/domain.rs

```diff
@@ -6,34 +6,90 @@ type Name = String;
 #[derive(Debug, Clone)]
 pub struct Domain {
     pub name: Name,
-    pub accounts: HashMap<Id, Account>,
+    pub accounts: HashMap<<Account as Identifiable>::Id, Account>,
+    pub assets: HashMap<<Asset as Identifiable>::Id, Asset>,
 }
 
 impl Domain {
     pub fn new(name: Name) -> Self {
         Domain {
             name,
             accounts: HashMap::new(),
+            assets: HashMap::new(),
         }
     }
 }
 
+impl Identifiable for Domain {
+    type Id = Name;
+}
+
 pub mod isi {
     use super::*;
-    use crate::isi::Contract;
-    use iroha_derive::{IntoContract, Io};
+    use crate::isi::Register;
+    use iroha_derive::*;
     use parity_scale_codec::{Decode, Encode};
 
-    /// The purpose of create domain command is to make new domain in Iroha network, which is a
-    /// group of accounts.
-    #[derive(Clone, Debug, PartialEq, Io, IntoContract, Encode, Decode)]
-    pub struct CreateDomain {
-        pub domain_name: String,
+    #[derive(Clone, Debug, Io, Encode, Decode)]
+    pub enum DomainInstruction {
+        RegisterAccount(Name, Account),
+        RegisterAsset(Name, Asset),
+    }
+
+    impl DomainInstruction {
+        pub fn execute(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
+            match self {
+                DomainInstruction::RegisterAccount(domain_name, account) => {
+                    Register::new(account.clone(), domain_name.clone()).execute(world_state_view)
+                }
+                DomainInstruction::RegisterAsset(domain_name, asset) => {
+                    Register::new(asset.clone(), domain_name.clone()).execute(world_state_view)
```

Comment:
I might have missed it, but where is the check performed that an asset with this name doesn't already exist?